Source/WebCore/PAL/ThirdParty/libavif/ThirdParty/dav1d/src/thread_task.c

```diff
@@ -453,6 +453,7 @@ static inline void delayed_fg_task(const Dav1dContext *const c,
         }
         row = atomic_fetch_add(&ttd->delayed_fg.progress[0], 1);
         int done = atomic_fetch_add(&ttd->delayed_fg.progress[1], 1) + 1;
+        (void)done; // placate clang static analyzer.
```

Comment:
Better:

```suggestion
        UNUSED_VARIABLE(done);
```

Reply:
> We need to change the upstream in ThirdParty code instead, right? (Otherwise, upstream import will undo the changes)
My thinking is that whoever does the merge / cherrypick from upstream, they will be conscientious to not destroy any bug fixes that we have done.  But if the only difference in the code are these static analyzer changes I made, I'm  fine with losing them and having to look for a more permanent solution later if the person took a short cut and undid all this.

Right now, my main goal is to thin out the analyzer complaints so that we can eventually get to the point of being able to see if there are any real bugs lying in there.  Right now, the sheer number of analyzer complaints are way to high for a human to parse.

>Better? Or we could write a macro only for the clang static analyzer that is an empty statement otherwise.
Is `UNUSED_VARIABLE` actually available to use in 3rd party code?  Looks like ANGLE has `ANGLE_UNUSED_VARIABLE`.  webrtc has nothing.  Don't know about all the other encoder / decoders.  I'm not sure if we can use our WTF macros in 3p code (because of dependencies).  Alternatively, we can try to make variants of the macro for all the pieces of code.

Do you have a strong preference?

Reply:
Oops, I didn't realize this as in third-party code.  You are correct that we should not use WTF macros here.

Although it's more work, I suspect we may want to split out the ANGLE and libwebrtc changes from this PR as @kkinnunen-apple prefers to land fixes upstream and them merge them back for ANGLE, and libwebrtc changes can get lost if they don't use `#ifdef WEBRTC_WEBKIT_BUILD`/`#endif` macros around the Apple-changed code.